docs/VERSIONING.md

```diff
@@ -0,0 +1,96 @@
+
+The direct dependencies between the submodules are as follows:
+
+```
+             ┌──────┐
+      ┌──────┤ core ├─────────┐
+      │      └──────┘         │
+   ┌──▼─┐                ┌────▼───┐
+   │ db ├─────┐          │ client │
+   └─┬──┘     │          └────┬───┘
+┌────▼────┐   │            ┌──▼──┐
+│ routers │   │   ┌────────┤ api │
+└─────────┘   │   │        └──┬──┘
+            ┌─▼───▼─┐      ┌──▼──┐
+            │ tasks │      │ cli │
+            └───────┘      └─────┘
+```
+
+
+### Versioning Strategy
+
+- Currently, we employ a versioning format of v0.<major>.<minor>.
+- We only support using the same version across all `diracx-*` packages.
+- The client and server versions can vary, but the client version is always communicated to the server. Incompatibly old client versions may lead to request rejections, prompting a client update.
+
+## Container Images
+
+DiracX utilizes a structured approach to containerization:
+
+1. **Base Image**:
+   - All container images start from `diracx/base`.
+
+2. **Specialized Base Images**:
+   - `diracx/services-base`
+   - `diracx/tasks-base`
+   - `diracx/client-base`
+
+3. **Image Versioning and Building**:
+   - Images are built periodically (e.g., every Monday) and tagged as `YYYY.MM.DD.P`.
+   - A DiracX release triggers the creation of new `DiracXService`, `diracx/tasks`, and `diracx/client` images, based on specific `diracx/base` tags.
+   - This approach ensures stability in production environments.
+   - For testing purposes, the `latest` base images are used, with dependencies installed via `pip install`.
+
+See this diagram for an example of how this looks in practice:
+
+```
+                       ┌──────────────────────────┐
+                 ┌─────┤ diracx/base:YYYY.MM.DD.P ├─────┐
+                 │     └──────────────────────────┘     │
+                 │                                      │
+┌────────────────▼──────────────────┐  ┌────────────────▼───────────────┐
+│ diracx/services-base:YYYY.MM.DD.P │  │ diracx/tasks-base:YYYY.MM.DD.P │
+└────────────────┬──────────────────┘  └────────────────┬───────────────┘
+                 │                                      │
+     ┌───────────▼────────────┐              ┌──────────▼──────────┐
+     │ diracx/services:v0.X.Y │              │ diracx/tasks:v0.X.Y │
+     └────────────────────────┘              └─────────────────────┘
+
+```
+
+### Dependencies
+
+- There is a noted duplication between `setup.cfg` and `environment.yaml`.
+- The `diracx/base` image is built from a Dockerfile with `environment.yml`, primarily defining the Python version.
+- The `diracx/services-base` and `diracx/tasks-base` images extend `diracx/base` with additional Dockerfiles and `environment.yml`, tailored to their specific needs.
+- The `diracx/services` and `diracx/tasks` images are further built upon their respective base images, adding necessary diracx packages through `pip install --no-dependencies`.
```

Comment:
is "images" the right name here?

Reply:
I think so. What else do you have in mind ?

Reply:
It's probably me getting confused: are `diracx/services` and `diracx/tasks` docker images?

Reply:
Yes